server/system/types/notification.go

```diff
@@ -0,0 +1,98 @@
+package types
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"time"
+
+	"github.com/cortezaproject/corteza/server/pkg/filter"
+)
+
+type (
+	Notification struct {
+		ID uint64 `json:"notificationID,string"`
+
+		Kind   NotificationKind   `json:"kind"`
+		Config NotificationConfig `json:"config"`
+
+		Recipient uint64    `json:"recipient,string"`
+		CreatedBy uint64    `json:"createdBy,string"`
+		CreatedAt time.Time `json:"createdAt,omitempty"`
+
+		ReadAt    *time.Time `json:"readAt"`
+		UpdatedAt *time.Time `json:"updatedAt,omitempty"`
+		DeletedAt *time.Time `json:"deletedAt,omitempty"`
+	}
+
+	NotificationFilter struct {
+		NotificationID []uint64           `json:"notificationID"`
+		Kind           []NotificationKind `json:"kind"`
+		Recipient      uint64             `json:"recipient,string"`
+		Read           filter.State       `json:"read"`
+		Deleted        filter.State       `json:"deleted"`
+
+		Check func(*Notification) (bool, error) `json:"-"`
+
+		filter.Sorting
+		filter.Paging
+	}
+
+	NotificationKind string
+
+	// NotificationConfig is a flexible configuration container for notification data
+	NotificationConfig struct {
+		Simple SimpleNotificationConfig `json:"simple"`
+		Record RecordNotificationConfig `json:"record"`
+	}
+
+	// SimpleNotificationConfig defines the structure for a simple notification
+	SimpleNotificationConfig struct {
+		Title       string `json:"title"`
+		Description string `json:"description"`
+	}
+
+	// RecordNotificationConfig defines the structure for a record notification
+	RecordNotificationConfig struct {
+		Title       string `json:"title"`
+		Description string `json:"description"`
+		ModuleID    uint64 `json:"moduleID,string"`
+		NamespaceID uint64 `json:"namespaceID,string"`
+		RecordID    uint64 `json:"recordID,string"`
+		OpenMode    string `json:"openMode,omitempty"` // "modal", "newTab", or "sameTab" (default)
```

Comment:
Nooo, do this instead

https://github.com/cortezaproject/corteza/blob/2024.9.x/server/system/types/template.go#L22
https://github.com/cortezaproject/corteza/blob/2024.9.x/server/system/types/template.go#L14
https://github.com/cortezaproject/corteza/blob/2024.9.x/server/system/types/template.go#L73